src/providers/TodosContext.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+import { Status } from '../utils/status';
+import { useLocalStorage } from '../hook/useLocalStorage';
+import { Todo } from '../utils/todo';
+
+interface TodosContextProps {
+  todos : Todo [];
+  addTodo: (title: string) => void;
+  toggleTodo: (id: number) => void;
+  deleteTodo: (id: number) => void;
+  editTodo: (id: number, title: string) => void;
+  allCompleted: () => void;
+  filteredTodos: Todo[];
+  setFilter: (string: Status) => void;
+  deleteCompetedTodos: () => void;
+}
+
+export const TodosContext = React.createContext<TodosContextProps>(
+  {} as TodosContextProps,
+);
+
+type Prop = {
+  children: React.ReactNode,
+};
+
+export const TodosProvider: React.FC<Prop> = ({ children }) => {
```

Comment:
let's create here our own hook
```
export const useTodo = ():TodosContextProps => React.useContext(TodosContext);
```
it will help us to avoid redundant import